FWCore/ParameterSet/bin/edmWriteConfigs.cpp

```diff
@@ -136,18 +138,33 @@ namespace {
     }
   }
 
-  void writeModulesFile() {
+  bool open_temp(std::string& path, std::ofstream& f) {
+    path += "/XXXXXX";
+    path += '\0';  // ensure that the string is null-terminated
+    int fd = mkstemp(path.data());
+    if (fd != -1) {
+      f.open(path.c_str(), std::ios_base::trunc | std::ios_base::out);
+      close(fd);
+      return true;
+    }
+    return false;
+  }
+
+  bool writeModulesFile() {
```

Comment:
With exceptions the function signature can return to
```suggestion
  void writeModulesFile() {
```